bindings/electron/src/lib.rs

```diff
@@ -10,6 +10,37 @@ lazy_static::lazy_static! {
         Mutex::new(tokio::runtime::Runtime::new().expect("Cannot start tokio runtime for libparsec"));
 }
 
+static BACKTRACE: Mutex<Option<Backtrace>> = Mutex::new(None);
+
+macro_rules! capture_backtrace {
+    ($tx: ident, $($tt: tt)*) => {
+        {
+            std::panic::set_hook(Box::new(|_| {
+                *crate::BACKTRACE.lock().expect("Mutex is poisoned") = Some(std::backtrace::Backtrace::capture());
+            }));
+
+            let ret = $($tt)*;
+
+            let _ = $tx.send(());
+
+            ret
+        }
+    };
+}
+
+macro_rules! handle_backtrace {
+    ($rx: ident, $cx: ident) => {
+        let _ = $rx.recv();
+
+        if let Some(backtrace) = crate::BACKTRACE.lock().expect("Mutex is poisoned").take() {
+            return $cx.throw_type_error(format!("Program panicked:\n{backtrace}"));
```

Comment:
```suggestion
            return $cx.throw_error(format!("Program panicked:\n{backtrace}"));
```

We don't know for sure that panic are caused by a type error

Reply:
True, I didn't search for others `errors`